### PR TITLE
Update Documents For speller In Ga 

### PR DESCRIPTION
**Pull request by ZChristine**

### CODE REVIEW

**Comment by Lauragra** [comment]
@ZChristine, can you please share the roadmap feature number that's associated with this PR? Thanks.

File: api-reference/v1.0/resources/alterationresponse.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  ],
+  "@odata.type": "microsoft.graph.alterationResponse",
+  "baseType": null
+}-->
+
+```json
+{
+  "originalQueryString": "String",
+  "queryAlteration": "String",
+  "queryAlterationType": "suggestion"
```
Comment (JarbasHorst) [inline api-reference/v1.0/resources/alterationresponse.md:41]: For consistency, the JSON representation should not contain examples. In this case, please replace "suggestion" with "String".

File: api-reference/v1.0/resources/alteredquerytoken.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  "optionalProperties": [
+
+  ],
+  "@odata.type": "microsoft.graph.alteredQueryToken",
+  "baseType": null
+}-->
+
+```json
+{
+  "offset": 0,
```
Comment (JarbasHorst) [inline api-reference/v1.0/resources/alteredquerytoken.md:39]: For consistency, the JSON representation should not contain examples. In this case, the value for **offset** and **length** should be "Int32".

File: api-reference/v1.0/resources/resulttemplate.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  "optionalProperties": [
+
+  ],
+  "@odata.type": "microsoft.graph.resultTemplate",
+  "baseType": null
+}-->
+
+
+```json
+{
```
Comment (JarbasHorst) [inline api-reference/v1.0/resources/resulttemplate.md:39]: For consistency, please format this JSON object.

File: api-reference/v1.0/resources/resulttemplateoption.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+  "optionalProperties": [
+
+  ],
+  "@odata.type": "microsoft.graph.resultTemplateOption",
+  "baseType": null
+}-->
+
+```json
+ {
+    "enableResultTemplate": true
```
Comment (JarbasHorst) [inline api-reference/v1.0/resources/resulttemplateoption.md:38]: For consistency, the JSON representation should not contain examples. In this case, please replace "true" with "Boolean". Suggested revision:

- "enableResultTemplate": "Boolean"

File: api-reference/v1.0/resources/resulttemplate.md [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+  "resultTemplateId": {
+                    "displayName": "String",
+                    "body": "Json schema"
```
Comment (JarbasHorst) [inline api-reference/v1.0/resources/resulttemplate.md:42]: Please, replace _Json schema_ with a value that describes the JSON representation of this property.  Suggested revision:

```
"body": {
    "@odata.type": "microsoft.graph.Json"
  }
```

File: api-reference/v1.0/resources/searchalteration.md [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+Namespace: microsoft.graph
+
+Provides the details of search alteration for spelling correction.
+
+## Properties
+
+| Property     | Type        | Description |
+|:-------------|:------------|:------------|
+|alteredQueryString|String| Defines the altered query string with spelling correction.|
+|alteredHighlightedQueryString|String| Defines the altered highlighted query string with spelling correction. The annotation around the corrected segment is (\ue000, \ue001)|
```
Comment (JarbasHorst) [inline api-reference/v1.0/resources/searchalteration.md:21]: Please, add a period at the end of the sentence. Also, I recommend you to highlight the corrected segment and remove the parentheses if possible. Suggested revision:

- ...the corrected segment is \`\ue000, \ue001\`.

File: api-reference/v1.0/resources/resulttemplateoption.md [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+
+Namespace: microsoft.graph
+
+Provides the search result templates options for render connectors search results.
+
+## Properties
+
+| Property     | Type        | Description |
+|:-------------|:------------|:------------|
+|enableResultTemplate|Boolean|Indicates whether search display layouts are enabled. If enabled, the user will get the result template to render the search results content in the **resultTemplates** property of the [response](/graph/api/resources/searchresponse?view=graph-rest-beta&preserve-view=true). The result template is based on [Adaptive Cards](https://adaptivecards.io/). This property is optional.|
```
Comment (JarbasHorst) [inline api-reference/v1.0/resources/resulttemplateoption.md:20]: For simplicity and consistency, I recommend you to replace "This property is optional." with "Optional.". Also, the URL for response refers to a beta resource topic; however, it should now refer to the v1.0 file.

File: api-reference/v1.0/resources/resulttemplateoption.md [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+author: "cristianv-ms"
+ms.prod: "search"
+doc_type: "resourcePageType"
+---
+
+# resultTemplateOption resource type
+
+Namespace: microsoft.graph
+
+Provides the search result templates options for render connectors search results.
```
Comment (JarbasHorst) [inline api-reference/v1.0/resources/resulttemplateoption.md:14]: For readability and clarity, please rephrase this sentence. Suggested revision:

- Provides the search result template options to render search results from connectors.

Please, use the same description also in line 3.

File: api-reference/v1.0/resources/search-api-overview.md [hunk 9]
```diff
@@ -132,6 +134,22 @@ Aggregations are currently supported for any refinable property on the following
...
+
+See [Request spelling correction](/graph/search-concept-speller) for examples that show how to use spelling corrections.
+
+## Search display layout
+
+The search API allows you to render search results from [connectors](/microsoftsearch/connectors-overview), by using the display layout or result template configured by the IT admin for each connector. The result templates are [Adaptive Cards](https://adaptivecards.io/), which are a semantically meaningful combination of layout and data.
+
+To get the result template in the [searchresponse](searchresponse.md), you have to set **true** the **enableResultTemplate** property, defined in the [resultTemplateOptions](./resulttemplateoption.md), in the [searchRequest](./searchrequest.md). The response includes a **resultTemplateId** for every [search hit](./searchhit.md), which maps to one of the display layouts included in the **resultTemplates** dictionary that is included in the response.
+
+See [Use search display layout](/graph/search-concept-display-layout) for examples.
```
Comment (JarbasHorst) [inline api-reference/v1.0/resources/search-api-overview.md:151]: For consistency and readability, we introduce the subject before we provide the link. In this case, I recommend: 

- For examples that show how to render search results, see \[Use search display layout\](/graph/search-concept-display-layout).

File: api-reference/v1.0/resources/search-api-overview.md [hunk 10]
```diff
@@ -132,6 +134,22 @@ Aggregations are currently supported for any refinable property on the following
 
 See [refine search results](/graph/search-concept-aggregation) for examples that show using aggregation to enhance and narrow down search results.
 
+## Request spelling correction
+
+Spelling correction is a popular way to handle mismatches between typos in a user query and the correct words in matched contents. When typos are detected in the original user query, you can get the search result either for the original user query or the corrected alternate query. You can also get the spelling correction information for typos in the **queryAlterationResponse** property of the [searchresponse](searchresponse.md).
+
+In the request body of the [query](/graph/api/search-query?view=graph-rest-beta&preserve-view=true) method, specify the **queryAlterationOptions** that should be applied to the query for spelling corrections. The description of **queryAlterationOptions** is defined in the [searchAlterationOptions](./searchalterationoptions.md).
+
+See [Request spelling correction](/graph/search-concept-speller) for examples that show how to use spelling corrections.
```
Comment (JarbasHorst) [inline api-reference/v1.0/resources/search-api-overview.md:143]: For consistency and readability, we introduce the subject before we provide the link. In this case, I recommend: 

- For examples that show how to use spelling corrections, see \[Request spelling correction\](/graph/search-concept-speller).

File: api-reference/v1.0/resources/search-api-overview.md [hunk 11]
```diff
@@ -132,6 +134,22 @@ Aggregations are currently supported for any refinable property on the following
 
 See [refine search results](/graph/search-concept-aggregation) for examples that show using aggregation to enhance and narrow down search results.
```
Comment (JarbasHorst) [inline api-reference/v1.0/resources/search-api-overview.md:135]: For consistency and readability, we introduce the subject before we provide the link. In this case, I recommend: 

- For examples that show how to use aggregation to enhance and narrow down search results, see \[Refine search results\](/graph/search-concept-aggregation).

File: api-reference/v1.0/resources/searchalteration.md [hunk 12]
```diff
@@ -0,0 +1,43 @@
...
+|alteredQueryTokens|[alteredQueryToken](alteredquerytoken.md) collection| Represents changed segments with respect to original query.|
```
Comment (JarbasHorst) [inline api-reference/v1.0/resources/searchalteration.md:22]: For consistency with the alteredQueryTokens resource topic, I recommend you to replace "with respect to original query" with "related to an original user query".

File: api-reference/v1.0/resources/searchalteration.md [hunk 13]
```diff
@@ -0,0 +1,43 @@
...
+# searchAlteration resource type
+
+Namespace: microsoft.graph
+
+Provides the details of search alteration for spelling correction.
+
+## Properties
+
+| Property     | Type        | Description |
+|:-------------|:------------|:------------|
```
Comment (JarbasHorst) [inline api-reference/v1.0/resources/searchalteration.md:19]: For consistency and to improve scanning, properties in the Properties table should be in alphabetical order. Please, alphabetize the list.

File: api-reference/v1.0/resources/searchrequest.md [hunk 14]
```diff
@@ -22,17 +22,20 @@ The JSON blob contains the types of resources expected in the response, the unde
...
 |from|Int32|Specifies the offset for the search results. Offset 0 returns the very first result. Optional.|
 |query|[searchQuery](searchquery.md)|Contains the query terms. Required.|
+|resultTemplateOptions|[resultTemplateOption](resultTemplateOption.md) collection|Provides the search result templates options to render search results from connectors.|
 |size|Int32|The size of the page to be retrieved. Optional.|
 |sortProperties|[sortProperty](sortProperty.md) collection|Contains the ordered collection of fields and direction to sort results. There can be at most 5 sort properties in the collection. Optional.|
+|resultTemplateOptions|[resultTemplateOption](resultTemplateOption.md) collection|Provides the search result templates options for rendering connectors search results.|
+|queryAlterationOptions|[searchAlterationOptions](searchalterationoptions.md)|Query alteration options formatted in a JSON blob that contains two optional flags related to spelling correction. Optional. |
```
Comment (JarbasHorst) [inline api-reference/v1.0/resources/searchrequest.md:38]: For consistency, please move the property **queryAlterationOptions** to come after **query**.

File: api-reference/v1.0/resources/searchresponse.md [hunk 15]
```diff
@@ -19,6 +19,8 @@ Represents results from a search query, and the terms used for the query.
 |:-------------|:------------|:------------|
 |hitsContainers|[searchHitsContainer](searchhitscontainer.md) collection|A collection of search results.|
 |searchTerms|String collection|Contains the search terms sent in the initial search query.|
+|resultTemplates|[resultTemplate](resultTemplate.md) collection|A dictionary of resultTemplateIds and associated values, which include the name and JSON schema of the result templates.
```
Comment (JarbasHorst) [inline api-reference/v1.0/resources/searchresponse.md:22]: For consistency, **resultTemplate.md** must be in lower case format. Also, please apply bold formatting to resultTemplateIds.

File: api-reference/v1.0/resources/searchresponse.md [hunk 16]
```diff
@@ -19,6 +19,8 @@ Represents results from a search query, and the terms used for the query.
...
+|queryAlterationResponse|[alterationResponse](alterationResponse.md)|Provides details of query alteration response for spelling correction.|
```
Comment (JarbasHorst) [inline api-reference/v1.0/resources/searchresponse.md:23]: For consistency with the alterationResponse topic, I recommend you to use "Provides information related to spelling corrections in the alteration response." as the description.

File: changelog/Microsoft.MicrosoftSearch.json [hunk 17]
```diff
@@ -329,6 +329,136 @@
       "CreatedDateTime": "2021-09-13T08:21:04.628851Z",
       "WorkloadArea": "Search",
       "SubArea": ""
+    },
+    {
+      "ChangeList": [
+        {
```
Comment (JarbasHorst) [inline changelog/Microsoft.MicrosoftSearch.json:335]: Is this changelog entry correct? If not, please remove this entry from this changelog record.

File: changelog/Microsoft.MicrosoftSearch.json [hunk 18]
```diff
@@ -329,6 +329,136 @@
...
+          "ChangeType": "Addition",
+          "Description": "Added the **searchAlterationType** enumeration type.",
+          "Target": "searchAlterationType"
+        },
+        {
+          "Id": "fca2af31-c9df-449e-b848-ecb564518c30",
+          "ApiChange": "Resource",
+          "ChangedApiName": "alterationResponse",
+          "ChangeType": "Addition",
+          "Description": "Added the [alterationResponse](https://docs.microsoft.com/en-us/graph/api/resources/alterationResponse?view=graph-rest-1.0) resource type.",
```
Comment (JarbasHorst) [inline changelog/Microsoft.MicrosoftSearch.json:356]: GLOBAL: Please apply this change to _all_ **Description** properties in this changelog record.

- URLs must be provided in lower case format. In this case, please replace "alterationResponse" with "alterationresponse" => ...api/resources/alterationresponse?view...

File: changelog/Microsoft.MicrosoftSearch.json [hunk 19]
```diff
@@ -329,6 +329,136 @@
...
+        },
+        {
+          "Id": "fca2af31-c9df-449e-b848-ecb564518c30",
+          "ApiChange": "Resource",
+          "ChangedApiName": "resultTemplate",
+          "ChangeType": "Addition",
+          "Description": "Added the [resultTemplate](https://docs.microsoft.com/en-us/graph/api/resources/resultTemplate?view=graph-rest-1.0) resource type.",
+          "Target": "resultTemplate"
+        },
+        {
```
Comment (JarbasHorst) [inline changelog/Microsoft.MicrosoftSearch.json:375]: This resource topic isn't part of this PR. Please, either add the resultTemplateDictionary resource topic to this PR or remove this entry.

File: changelog/Microsoft.MicrosoftSearch.json [hunk 20]
```diff
@@ -329,6 +329,136 @@
...
+          "ChangeType": "Addition",
+          "Description": "Added the [searchAlterationOptions](https://docs.microsoft.com/en-us/graph/api/resources/searchAlterationOptions?view=graph-rest-1.0) resource type.",
+          "Target": "searchAlterationOptions"
+        },
+        {
+          "Id": "fca2af31-c9df-449e-b848-ecb564518c30",
+          "ApiChange": "Property",
+          "ChangedApiName": "resultTemplateId",
+          "ChangeType": "Addition",
+          "Description": "Added the **resultTemplateId** property to [searchHit](https://docs.microsoft.com/en-us/graph/api/resources/searchHit?view=graph-rest-1.0) resource.",
```
Comment (JarbasHorst) [inline changelog/Microsoft.MicrosoftSearch.json:412]: Please, add "the" before "[searchHit]".

File: changelog/Microsoft.MicrosoftSearch.json [hunk 21]
```diff
@@ -329,6 +329,136 @@
...
+          "Target": "searchHit"
+        },
+        {
+          "Id": "fca2af31-c9df-449e-b848-ecb564518c30",
+          "ApiChange": "Property",
+          "ChangedApiName": "queryAlterationOptions",
+          "ChangeType": "Addition",
+          "Description": "Added the **queryAlterationOptions** property to [searchRequest](https://docs.microsoft.com/en-us/graph/api/resources/searchRequest?view=graph-rest-1.0) resource.",
```
Comment (JarbasHorst) [inline changelog/Microsoft.MicrosoftSearch.json:420]: Please, add "the" before "[searchRequest]".

File: changelog/Microsoft.MicrosoftSearch.json [hunk 22]
```diff
@@ -329,6 +329,136 @@
...
+          "Target": "searchRequest"
+        },
+        {
+          "Id": "fca2af31-c9df-449e-b848-ecb564518c30",
+          "ApiChange": "Property",
+          "ChangedApiName": "resultTemplateOptions",
+          "ChangeType": "Addition",
+          "Description": "Added the **resultTemplateOptions** property to [searchRequest](https://docs.microsoft.com/en-us/graph/api/resources/searchRequest?view=graph-rest-1.0) resource.",
```
Comment (JarbasHorst) [inline changelog/Microsoft.MicrosoftSearch.json:428]: Please, add "the" before "[searchRequest]".

File: changelog/Microsoft.MicrosoftSearch.json [hunk 23]
```diff
@@ -329,6 +329,136 @@
...
+          "Target": "searchRequest"
+        },
+        {
+          "Id": "fca2af31-c9df-449e-b848-ecb564518c30",
+          "ApiChange": "Property",
+          "ChangedApiName": "queryAlterationResponse",
+          "ChangeType": "Addition",
+          "Description": "Added the **queryAlterationResponse** property to [searchResponse](https://docs.microsoft.com/en-us/graph/api/resources/searchResponse?view=graph-rest-1.0) resource.",
```
Comment (JarbasHorst) [inline changelog/Microsoft.MicrosoftSearch.json:436]: Please, add "the" before "[searchResponse]".

File: changelog/Microsoft.MicrosoftSearch.json [hunk 24]
```diff
@@ -329,6 +329,136 @@
...
+          "Target": "searchResponse"
+        },
+        {
+          "Id": "fca2af31-c9df-449e-b848-ecb564518c30",
+          "ApiChange": "Property",
+          "ChangedApiName": "resultTemplates",
+          "ChangeType": "Addition",
+          "Description": "Added the **resultTemplates** property to [searchResponse](https://docs.microsoft.com/en-us/graph/api/resources/searchResponse?view=graph-rest-1.0) resource.",
```
Comment (JarbasHorst) [inline changelog/Microsoft.MicrosoftSearch.json:444]: Please, add "the" before "[searchResponse]".

File: changelog/Microsoft.MicrosoftSearch.json [hunk 25]
```diff
@@ -329,6 +329,136 @@
...
+          "Target": "searchResponse"
+        },
+        {
```
Comment (JarbasHorst) [inline changelog/Microsoft.MicrosoftSearch.json:447]: Please, remove this entry from this changelog record.

**Comment by ZChristine** [comment]
> @ZChristine, can you please share the roadmap feature number that's associated with this PR? Thanks.

sure. primary oncall last week. will share it later.

File: changelog/Microsoft.MicrosoftSearch.json (hunk 25 above)
Comment (ZChristine) [reply changelog/Microsoft.MicrosoftSearch.json:447]: i resolve other comments. why do we need to remove this entry. the change log is auto generated when csdl change is merged.

File: api-reference/beta/api/accesspackage-filterbycurrentuser.md [hunk 26]
```diff
@@ -38,7 +38,7 @@ The following table shows the parameters that can be used with this function.
 
 |Parameter|Type|Description|
 |:---|:---|:---|
-|on|[accessPackageFilterByCurrentUserOptions](../resources/accesspackage-accesspackagefilterbycurrentuseroptions.md)|The list of current user options that can be used to filter on the access packages list. The allowed value is `allowedRequestor`.|
+|on|accessPackageFilterByCurrentUserOptions|The list of current user options that can be used to filter on the access packages list. The allowed value is `allowedRequestor`.|
 
 - `allowedRequestor` is used to get the `accessPackage` objects for which the signed-in user is allowed to submit access requests. The resulting list includes all access packages that can be requested by the caller across all catalogs.
```
Comment (ZChristine) [inline api-reference/beta/api/accesspackage-filterbycurrentuser.md:44]: @JarbasHorst  I have updated the change based on your comment. This branch is merged with error and it cannot be override forcely. can we pls remove to https://github.com/microsoftgraph/microsoft-graph-docs/pull/15840?

Comment (JarbasHorst) [reply api-reference/beta/api/accesspackage-filterbycurrentuser.md:44]: I checked internally and you can please close PR 15636. Let's continue the content review work from PR 15840 as you suggested. 

I noticed that there are still 37 open comments that haven't been addressed in this PR. I added them again to PR 15840. Please, verify my comments in PR 15840.

**Pull request closed without merging**